corehq/apps/userreports/models.py

```diff
@@ -1486,10 +1486,13 @@ def __init__(self, expression, factory_context):
         self.wrapped_expression = None
 
     def __call__(self, *args, **kwargs):
+        self.unwrap()
+        return self.wrapped_expression(*args, **kwargs)
+
+    def unwrap(self):
```

Comment:
> Or possibly make it a `django.utils.functional.cached_property` named `wrapped_expression`

I like this suggestion too - it's used in one place to mutate the class and in the other place for the return value.  Making this a cached property would allow it to be used the same way in both contexts.

Reply:
1160e29e9f6b520f33fca98cabcc162fd7303498